src/server/pkg/deploy/cmds/cmds.go

```diff
@@ -434,11 +432,24 @@ If <object store backend> is \"s3\", then the arguments are:
 		}
 		defer c.Close()
 
-		_, err = c.DeployStorageSecret(context.Background(), &deployclient.DeployStorageSecretRequest{
-			Secrets: data,
-		})
+		manifest := getEncoder(outputFormat)
+		err = assets.WriteSecret(manifest, data, opts)
 		if err != nil {
-			return fmt.Errorf("error deploying storage secret to pachd: %v", err)
+			return err
+		}
+		if dryRun {
+			_, err := os.Stdout.Write(manifest.Buffer().Bytes())
+			return err
+		}
+
+		io := cmdutil.IO{
+			Stdin:  manifest.Buffer(),
+			Stdout: os.Stdout,
+			Stderr: os.Stderr,
+		}
+		// we set --validate=false due to https://github.com/kubernetes/kubernetes/issues/53309
+		if err := cmdutil.RunIO(io, "kubectl", "apply", "-f", "-", "--validate=false", "--namespace", opts.Namespace); err != nil {
+			return err
```

Comment:
I think this does what we want, would you mind doing a manual test on this to make sure the secret we're getting looks right? In particular I'm pretty sure that `apply` merges the way we want it to, but I may be misunderstanding its docs in which case it would actually be overwriting the existing storage secret, which would be bad.

Reply:
i verified that the updated version only overwrites the pieces of data which get new values, and leaves the rest intact